### ISSUE TITLE
optional attributes

### ISSUE THREAD

**Issue by miguelski** [utkarshkukreti/markup.rs#14]
problem:
i am migrating a project from horrorshow currently there are some Option variables for example 
`let css: Option<String> = None;`
but i don't find a way to specify that if css is None don't create the html attribute
`div[class=css] `
creates
`<div class=""></div>`
but i want create when css is None
`<div></div>`
i know i can do an if conditional like this
```
@if let Option(css) = css{
div[class=css]
}else
{
div{}
}
```
but this is not practical when i have a lot of optional attributes

horrorshow have this sintaxis 
`div(class?=css) `
and i get what i want `<div></div>` when css is None

is there something similar that i missing? or it would be a new feature?

**Comment by ZaneHannanAU** [comment]
May be possible with bailout conditions div[class=css?] and have the codegen put it in its own

```rust
let _: Option<io::Result<()>> = {
  Some(writer.write("<div class=\"{}\">", css?))
}
```

but seems like a fair bit more work and codegen

**Comment by ZaneHannanAU** [comment]
Also

**Comment by ZaneHannanAU** [comment]
@ZaneHannanAU misread the original...isn't quite so simple. The class="" shouldn't be too bad outside of looking like php.

**Comment by miguelski** [comment]
reading the code it should work,
i make it work but if the variable is Option<string> we need to use as_ref() and if the variable is Option<& str> use * otherwise it doesn't work, same for false boolean if it is not used with * don't work.

![image](https://user-images.githubusercontent.com/72758941/118344208-c54a5380-b4f2-11eb-98e1-95af112bbe30.png)

**Comment by miguelski** [comment]
https://github.com/utkarshkukreti/markup.rs/blob/4dfc1275ffae79714539004c1762f3b3f9430989/markup-proc-macro/src/generate.rs#L137

i think the issue is in the above methods. `is_none` when used &Option is always false and `is_false` is always false if a &boolean is passed instead a boolean

**Comment by utkarshkukreti** [comment]
Thanks for reporting the issue and figuring out the solution @miguelski! Can you please try out the latest version from git to confirm this fixes your code before I make a release (I have added tests, but just to be sure)?

```toml
[dependencies]
markup = { git = "https://github.com/utkarshkukreti/markup.rs" }
```

**Comment by miguelski** [comment]
working for me 
![image](https://user-images.githubusercontent.com/72758941/118345785-6ee21280-b4fc-11eb-93fa-962a5bce9720.png)

**Comment by utkarshkukreti** [comment]
Awesome, published 0.12.1.
